lib/dart_convert/converted_complex_object.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:jsonexample/dart_convert/converted_simple_object.dart';
+
+class ConvertedComplexObject {
+  const ConvertedComplexObject({
+    this.aString,
+    this.anInt,
+    this.aDouble,
+    this.anObject,
+    this.aListOfStrings,
+    this.aListOfInts,
+    this.aListOfDoubles,
+    this.aListOfObjects,
+  });
+
+  final String aString;
+  final int anInt;
+  final double aDouble;
+  final ConvertedSimpleObject anObject;
+  final List<String> aListOfStrings;
+  final List<int> aListOfInts;
+  final List<double> aListOfDoubles;
+  final List<ConvertedSimpleObject> aListOfObjects;
+
+  factory ConvertedComplexObject.fromJson(Map<String, dynamic> json) {
+    if (json == null)
+      return null;
```

Comment:
dartfmt would fix this, too. We don't like hanging implicit blocks – either one line or add `{}`

Reply:
Fixed.